activerecord/lib/active_record/future_result.rb

```diff
@@ -51,20 +56,22 @@ def execute_or_skip
 
     def result
       execute_or_wait
-      if @error
-        raise @error
-      elsif canceled?
+      @event_buffer&.flush
+
+      if canceled?
```

Comment:
I changed the order here to avoid a flaky test. The cancelation test would sometimes fail because the query would be executed before we access the result.

But either way I think it makes sense to disallow accessing a canceled result, even if we have it.